ai-service/src/ai_service/project_ingestor.py

```
@@ -0,0 +1,60 @@
+
+def clone_github_repo(repo_url: str) -> str:
+    """
+    Clones a GitHub repo to a temporary directory.
+    Returns the path to the cloned directory.
+    """
+    clone_to = tempfile.mkdtemp()
+    subprocess.run(["git", "clone", repo_url, clone_to], check=True)
+    return clone_to
+
+
+def scan_code_files(root_dir: str) -> List[Dict]:
```

Ruff (0.12.2)

Use `list` instead of `List` for type annotation

Replace with `list`

(UP006)